### PR TITLE
Remove specific metadata in shared Bash remediations

### PR DESCRIPTION
**Pull request by matejak**
This PR improves the problematic situation with remediations for different products and the respective metadata.

When a rule has exactly one `shared.sh` file, then cases when remediation shouldn't be performed are extremely rare, and even then they should be handled by means of the more flexible jinja macros.

`accounts_password_warn_age_login_defs` and `accounts_minimum_age_login_defs` got their remediations that were specialized for no good reason unified.

Question: Is there any reason that e.g. https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/set_iptables_default_rule_forward/rule.yml had RHEV-only remediation? The history doesn't indicate it at all. (answered below - there is no reason).

Fixes: #7246

This PR has been separated from its Ansible counterpart, as Ansible remediations in the project are more tricky regarding product-specificity.

### CODE REVIEW

**Comment by yuumasato** [comment]
> Question: Is there any reason that e.g. https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/set_iptables_default_rule_forward/rule.yml had RHEV-only remediation? The history doesn't indicate it at all.

From https://github.com/ComplianceAsCode/content/commits/master/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/set_iptables_default_rule_forward/bash

It seems to me that it was not intentional.
The remediation was marked as rhel6 only, then the rhv4 was added.
But when rhel6 product was removed, the rhel6 marking went away and only rhv4 remained.
As the rule has no specific prodtype the remediation very likely applies to many products.

**Review by yuumasato — APPROVED** [review]
LGTM

**Comment by matejak** [comment]
I did the same with Ansible, and that will be it.

File: linux_os/guide/system/accounts/accounts-session/accounts_tmout/bash/shared.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# platform = Red Hat Enterprise Linux 7,Red Hat Enterprise Linux 8,Red Hat Virtualization 4,multi_platform_fedora,multi_platform_ol,multi_platform_wrlinux
+# platform = multi_platform_rhel,Red Hat Virtualization 4,multi_platform_fedora,multi_platform_ol,multi_platform_wrlinux
```
Comment (vojtapolasek) [inline linux_os/guide/system/accounts/accounts-session/accounts_tmout/bash/shared.sh:1]: Why not multi_platform_all here?

File: linux_os/guide/system/permissions/files/permissions_within_important_dirs/dir_ownership_library_dirs/ansible/shared.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# platform = multi_platform_sle
+# platform = multi_platform_all
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_within_important_dirs/dir_ownership_library_dirs/ansible/shared.yml:1]: This is originally for SLE only... is that OK?

File: linux_os/guide/system/permissions/files/permissions_within_important_dirs/dir_permissions_library_dirs/ansible/shared.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# platform = multi_platform_sle
+# platform = multi_platform_all
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_within_important_dirs/dir_permissions_library_dirs/ansible/shared.yml:1]: Also this is originally only for SLE.

File: linux_os/guide/system/permissions/files/permissions_within_important_dirs/dir_permissions_library_dirs/bash/shared.sh [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# platform = multi_platform_sle
+# platform = multi_platform_all
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_within_important_dirs/dir_permissions_library_dirs/bash/shared.sh:1]: Also this is SLE only.

File: linux_os/guide/system/software/integrity/software-integrity/aide/aide_verify_acls/ansible/shared.yml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# platform = multi_platform_sle
+# platform = multi_platform_all
```
Comment (vojtapolasek) [inline linux_os/guide/system/software/integrity/software-integrity/aide/aide_verify_acls/ansible/shared.yml:1]: Also this one originally SLE only?

File: linux_os/guide/system/software/integrity/software-integrity/aide/aide_verify_ext_attributes/ansible/shared.yml [hunk 6]
```diff
@@ -1,4 +1,4 @@
-# platform = multi_platform_sle
+# platform = multi_platform_all
```
Comment (vojtapolasek) [inline linux_os/guide/system/software/integrity/software-integrity/aide/aide_verify_ext_attributes/ansible/shared.yml:1]: Also this is SLE only

File: linux_os/guide/system/software/updating/clean_components_post_updating/bash/shared.sh [hunk 7]
```diff
@@ -1,4 +1,4 @@
-# platform = Red Hat Enterprise Linux 7,Red Hat Enterprise Linux 8,multi_platform_ol,multi_platform_rhv
+# platform = multi_platform_rhel,multi_platform_ol,multi_platform_rhv
```
Comment (vojtapolasek) [inline linux_os/guide/system/software/updating/clean_components_post_updating/bash/shared.sh:1]: Why not multi_platform_all?

File: linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_warn_age_login_defs/bash/shared.sh [hunk 8]
```diff
@@ -1,9 +1,11 @@
-# platform = Red Hat Enterprise Linux 7,Red Hat Enterprise Linux 8,Red Hat Virtualization 4
+# platform = multi_platform_all
 . /usr/share/scap-security-guide/remediation_functions
+declare var_accounts_password_warn_age_login_defs
 {{{ bash_instantiate_variables("var_accounts_password_warn_age_login_defs") }}}
 
 grep -q ^PASS_WARN_AGE /etc/login.defs && \
-  sed -i "s/PASS_WARN_AGE.*/PASS_WARN_AGE     $var_accounts_password_warn_age_login_defs/g" /etc/login.defs
```
Comment (vojtapolasek) [inline linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_warn_age_login_defs/bash/shared.sh:6]: Is this change intentional? It seems a bit out of scope of this PR.

**Review by vojtapolasek — CHANGES_REQUESTED** [review]
LGTM, at some places I just pointed out some misalignments, fell free to comment.

File: linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_warn_age_login_defs/bash/shared.sh (hunk 8 above)
Comment (matejak) [reply linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_warn_age_login_defs/bash/shared.sh:6]: Some of the remediations were duplicated for different product, so this one is a fusion of multiple similar ones.

File: linux_os/guide/system/software/updating/clean_components_post_updating/bash/shared.sh (hunk 7 above)
Comment (matejak) [reply linux_os/guide/system/software/updating/clean_components_post_updating/bash/shared.sh:1]: It is yum-specific.-

**Comment by matejak** [comment]
I can't answer the Ansible applicability questions before I take off for my PTO, so if you don't want to seek the answers by yourselves, then please split the PR and use only the first commit.

File: linux_os/guide/system/accounts/accounts-physical/screen_locking/smart_card_login/install_smartcard_packages/ansible/shared.yml [hunk 9]
```diff
@@ -1,4 +1,4 @@
-# platform = multi_platform_sle
```
[the remark this answers is not in the document]
Comment (ggbecker) [reply linux_os/guide/system/accounts/accounts-physical/screen_locking/smart_card_login/install_smartcard_packages/ansible/shared.yml:1]: This is definitely a problem. The rule.yml uses package installed template. This should either be reverted or the file should be split into two: sle12.yml and sle15.yml

File: linux_os/guide/system/permissions/files/permissions_within_important_dirs/dir_permissions_library_dirs/bash/shared.sh (hunk 4 above)
Comment (matejak) [reply linux_os/guide/system/permissions/files/permissions_within_important_dirs/dir_permissions_library_dirs/bash/shared.sh:1]: Sure, it reduces permissions of subdirectories according to the rule description.

File: linux_os/guide/system/accounts/accounts-session/accounts_tmout/bash/shared.sh (hunk 1 above)
Comment (matejak) [reply linux_os/guide/system/accounts/accounts-session/accounts_tmout/bash/shared.sh:1]: Good point, the rule has an opt-in prodtype, so there is no reason to be too careful here.

File: linux_os/guide/system/accounts/accounts-physical/screen_locking/smart_card_login/install_smartcard_packages/ansible/shared.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (cipherboy) [reply linux_os/guide/system/accounts/accounts-physical/screen_locking/smart_card_login/install_smartcard_packages/ansible/shared.yml]: @ggbecker @matejak Just as a note, in #6792 we allowed the generic `sle.yml` so you don't have to worry about duplicated content.

[the remark this answers is not in the document]
Comment (cipherboy) [reply linux_os/guide/system/accounts/accounts-physical/screen_locking/smart_card_login/install_smartcard_packages/ansible/shared.yml]: Now actually documented in #7291

**Comment by matejak** [comment]
I have split the more controversial Ansible remediations to another PR.

**Review by vojtapolasek — APPROVED** [review]
Looks good now, merging.
